app/transport/http/user.go

```diff
@@ -11,6 +11,10 @@ func (uh UserHandler) CreateUser() http.Handler {
 	return http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
 		usr := entities.User{}
 
+		if err := DecodeBody(req, &usr); err != nil {
```

Comment:
You don't need that function at all. Just put its logic here.
Having it as a separate function doesn't make code more readable.

Reply:
@vpakhuchyi  Having helper functions like decode, endode, respErr will make our code more DRY if we are going to propagate our handlers, isnt it?

Reply:
fix it in new commit

Reply:
@delveper you are right
However, there is no sense to move 2 line to a separate function cos these 2 lines may look more simple than a new function with input and output arguments.

Here we tried to encapsulate call to json.Decode().
Just a single line of code or 3 if we count errors handling. 

```
	        if err := json.NewDecoder(req.Body).Decode(&user); err != nil {
			return //someone have to work with errors
		}
		defer req.Body.Close()
```

Is it much easier to read than:
```
                if err := DecodeBody(req, usr); err != nil {
			return
		}
```

I don't think so. Moreover, `DecodeBody` is a custom function, so i have to read it to understand what is going on inside it. But `json.Decode` - is a standard one - i know it.


After all, you have to split code to function when
1. it's reusable (but this isn't) 
2. original function is too big, so you want to simplify it

In the future we will add more rules, but that's all for now.